src/Components/WebAssembly/WebAssembly.Authentication/src/Models/RemoteAuthenticationResult.cs

```diff
@@ -7,7 +7,7 @@ namespace Microsoft.AspNetCore.Components.WebAssembly.Authentication
     /// Represents the result of an authentication operation.
     /// </summary>
     /// <typeparam name="TState">The type of the preserved state during the authentication operation.</typeparam>
-    public class RemoteAuthenticationResult<TState> where TState : class
+    public class RemoteAuthenticationResult<TState> where TState : RemoteAuthenticationState
```

Comment:
```suggestion
    public class RemoteAuthenticationResult<TRemoteAuthenticationState> where TRemoteAuthenticationState : RemoteAuthenticationState
```

to be consistent with other APIs